ion/src/device/shared/drivers/timing.cpp

```diff
@@ -17,11 +17,16 @@ using namespace Device::Timing;
 void msleep(uint32_t ms) {
   // We decrease the AHB clock frequency to save power while sleeping.
   Device::Board::setClockFrequency(Device::Board::Frequency::Low);
+  // Change the systick frequency to compensate the KCLK frequency change
+  Device::Timing::changeSysTickFrequency(Ion::Device::Clocks::Config::HCLKLowFrequency);
```

Comment:
Call to `Device::Timing::changeSysTickFrequency` should be inside `Device::Board::setClockFrequency` implementation. As soon as the clock frequency is modified, systick has to be adjusted.